lightning/src/ln/chan_utils.rs

```diff
@@ -427,7 +427,10 @@ pub struct HTLCOutputInCommitment {
 	pub transaction_output_index: Option<u32>,
 }
 
-impl_writeable!(HTLCOutputInCommitment, HTLC_OUTPUT_IN_COMMITMENT_SIZE, {
+impl_writeable_len_match!(HTLCOutputInCommitment, {
+		{ HTLCOutputInCommitment { transaction_output_index: None, .. }, 1 + 8 + 4 + 32 + 1},
```

Comment:
Could do `HTLC_OUTPUT_IN_COMMITMENT_SIZE - 4` or update `HTLC_OUTPUT_IN_COMMITMENT_SIZE` to not include the `u32` size and adjust accordingly.